CMakeLists.txt

```diff
@@ -0,0 +1,100 @@
+project (imgui)
+cmake_minimum_required (VERSION 3.2)
+
+option (IMGUI_EXAMPLES "Build ImGui examples" ON)
+option (IMGUI_DEMO "Include the ImGui demo window implementation in library" ON)
+
+option (IMGUI_IMPL_SDL "Build the SDL implementation (only if supported)" ON)
+option (IMGUI_IMPL_METAL "Build the Metal implementation (only if supported)" ${APPLE})
+option (IMGUI_IMPL_OSX "Build the OSX implementation (only if supported)" ${APPLE})
+option (IMGUI_IMPL_WIN32 "Build the Win32 (native winapi) implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_GLFW "Build the GLFW implementation (only if supported)" ON)
+option (IMGUI_IMPL_GLUT "Build the GLUT implementation (only if supported)" ON)
+option (IMGUI_IMPL_OPENGL "Build the OpenGL implementation (only if supported)" ON)
+option (IMGUI_IMPL_OPENGL2 "Build the OpenGL 2 (legacy) implementation (only if supported)" ${IMGUI_IMPL_OPENGL})
+option (IMGUI_IMPL_DX9 "Build the DirectX 9 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX10 "Build the DirectX 10 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX11 "Build the DirectX 11 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX12 "Build the DirectX 12 implementation (only if supported)" ${WIN32})
```

Comment:
Instead of having a separate option for each implementation, could this just be replaced with a cached list variable of backends to be build? Having individual yes/no options is convenient when configuring cmake via a gui, but a nightmare when doing so on the command line. Keep in mind that if imgui is used via `add_subdirectory`  those aren't the only options a developer has to deal with.

Reply:
> Instead of having a separate option for each implementation, could this just be replaced with a cached list variable of backends to be build?

That's pretty unusual for a cmake project. I'd be hestiant to manage config that way.

> Having individual yes/no options is convenient when configuring cmake via a gui, but a nightmare when doing so on the command line.

I don't really agree; a few keystrokes difference doesn't really warrant changing the convention most properly written CMake configurations follow.

> Keep in mind that if imgui is used via `add_subdirectory` those aren't the only options a developer has to deal with.

I don't follow; could you explain?

Reply:
Here is an example of what I had in mind: 

https://github.com/boostorg/cmake/blob/efbbc015ba9de6367880a6a1e0e1ce0dd52fe371/include/BoostRoot.cmake#L29

Imho 

    cmake .. -DBUILD_VERSIONS=Foo;Bar;Boo

is much more elegant than 

    cmake .. -DBUILD_FOO=ON -DBUILD_BAR=ON -DBUILD_BAZ=OFF -DBUILD_BOO=ON

But well, just a suggestion.